surfactant/plugin/manager.py

```diff
@@ -65,6 +66,25 @@ def _register_plugins(pm: pluggy.PluginManager) -> None:
     for plugin in internal_plugins:
         pm.register(plugin)
 
+    config_manager = ConfigManager()
+
+    # Retrieve the current list of blocked plugins
+    current_blocked_plugins = config_manager.get("core", "disable_plugins", [])
+    for plugin_name in current_blocked_plugins:
+        # Check if the plugin is already blocked
+        if pm.is_blocked(plugin_name):
+            print(f"Plugin '{plugin_name}' is already disabled.")
+            continue
+
+        # Unregister the plugin
+        plugin = pm.unregister(name=plugin_name)
+        if plugin is None:
+            print(f"Plugin '{plugin_name}' not found.")
+            continue
+
+        # Block the plugin to prevent future registration
+        pm.set_blocked(plugin_name)
+
```

Comment:
Let's move all of this into its own disable plugins function that takes a pluggy.PluginManager as an argument, then call that function from in the get_plugin_manager function below (right after the call to _register_plugins.